### ISSUE TITLE
[Feature Request] allow a matrix to scalar multiply  a fraction in string

### ISSUE THREAD

**Issue by royqh1979** [sympy/sympy#17500]
Often we need to multiply a matrix by a fraction. 
But now the matrices in sympy don't support to multiply a fraction in string format. Codes like:
`M * "1/2"`
or
`"1/2" * M`
will raise a type error.

It would be awesome if scalar multiply a string fraction is allowed

**Comment by oscarbenjamin** [comment]
I don't think this is a good idea. We should allow strings in fewer places not more. You can use sympify/S to do this:
```
M * S("1/2")
```

**Comment by sylee957** [comment]
I think that you are able to build expression with strings like
```python3
>>> M = MatrixSymbol('M', 3, 3)
>>> MatMul(M, '1/2')
M*(1/2)
```

**Comment by oscarbenjamin** [comment]
I don't think that `MatMul(M, '1/2')` should work either. The problem is that sympify parses strings so many functions unintentionally accept strings because they sympify their arguments. We should disallow using sympify to parse strings or should stop using sympify (in favour of `_sympify`) internally.
